src/common.ts

```diff
@@ -68,7 +105,49 @@ export interface Transferable {
 }
 
 export interface Task {
-  readonly [kQueueOptions] : object | null;
+  readonly [kQueueOptions] : {} | null;
+  filename: string;
+  name: string;
+  taskId: number;
+}
+
+export abstract class AsynchronouslyCreatedResource {
+  onreadyListeners : (() => void)[] | null = [];
+
+  markAsReady () : void {
+    const listeners = this.onreadyListeners;
+    assert(listeners !== null);
+    this.onreadyListeners = null;
+    for (const listener of listeners) {
+      listener();
+    }
+  }
+
+  isReady () : boolean {
+    return this.onreadyListeners === null;
+  }
+
+  onReady (fn : () => void) {
```

Comment:
What about failures?

Reply:
Hmm, good point; it seems we immediately remove the worker without even calling `onReady`. This might be opaque for the Scheduler. Although I'm not sure how much value it will bring, anyways as the Scheduler just won't account it for its pool